drivers/cache/cache.go

```diff
@@ -0,0 +1,244 @@
+	cache, err := redis.Dial(redis.Config{})
+	if err != nil {
+		// Handle connection error
+	}
+
+	// Connect to the Cache database
+	db, err := cache.Dial(cache.Config{
+		Database: database,
+		Cache: 	  cache,
+	})
+	if err != nil {
+	    // Handle connection error
+	}
+
+	err := db.Setup()
+	if err != nil {
+	    // Handle setup error
+	}
+
+	// Set a value
+	err = db.Set("key", []byte("value"))
+	if err != nil {
+	    // Handle error
+	}
+
+	// Retrieve a value
+	value, err := db.Get("key")
+	if err != nil {
+	    // Handle error
+	}
+*/
+package cache
+
+import (
+	"errors"
+
+	"github.com/madflojo/hord"
+)
+
+// Config provides the configuration options for the Cache driver.
+type Config struct {
+	Database hord.Database
+	Cache    hord.Database
+}
+
+// Cache is used to store data in a look-aside caching pattern. It also satisfies the Hord database interface.
+type Cache struct {
+	data  hord.Database
+	cache hord.Database
+}
+
+var (
+	// ErrInvalidDatabase is returned when a database is nil.
+	ErrInvalidDatabase = errors.New("database cannot be nil")
+
+	// ErrInvalidCache is returned when a cache is nil.
+	ErrInvalidCache = errors.New("cache cannot be nil")
+)
+
+// Dial will create a new Cache driver using the provided Config. It will return an error if either the Database or Cache values in Config are nil.
+func Dial(cfg Config) (*Cache, error) {
+	if cfg.Database == nil {
+		return nil, ErrInvalidDatabase
+	}
+
+	if cfg.Cache == nil {
+		return nil, ErrInvalidCache
+	}
+
+	return &Cache{
+		data:  cfg.Database,
+		cache: cfg.Cache,
+	}, nil
+}
+
+// Setup will run the Setup function for both the database and the cache.
+func (db *Cache) Setup() error {
+	if db == nil || db.data == nil || db.cache == nil {
+		return hord.ErrNoDial
+	}
+
+	if err := db.data.Setup(); err != nil {
+		return err
+	}
+
+	if err := db.cache.Setup(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HealthCheck will run the HealthCheck function for both the database and the cache.
+func (db *Cache) HealthCheck() error {
+	if db == nil || db.data == nil || db.cache == nil {
+		return hord.ErrNoDial
+	}
+
+	dataErr := db.data.HealthCheck()
+	cacheErr := db.cache.HealthCheck()
+
+	if dataErr != nil {
+		return dataErr
+	} else if cacheErr != nil {
+		return cacheErr
+	}
+
+	return nil
+}
+
+// Get will get the data from the database. It uses a look-aside pattern to store the data in the cache if it is not already there.
+func (db *Cache) Get(key string) ([]byte, error) {
+	if db == nil || db.data == nil || db.cache == nil {
+		return nil, hord.ErrNoDial
+	}
+
+	// Check the cache first
+	data, err := db.cache.Get(key)
+	if (err != nil) && !errors.Is(err, hord.ErrNil) {
+		return nil, err
+	} else if !errors.Is(err, hord.ErrNil) {
+		return data, nil
+	}
+
+	// Check the data database
+	data, err = db.data.Get(key)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the cache
+	err = db.cache.Set(key, data)
+	if err != nil {
+		return data, err
```

Comment:
I wonder if wrapping the original driver error with an `ErrCacheError` or an `ErrDatabaseError` error type makes sense.

Today drivers return these errors, but wrapping them might give users more ideas on where the error was.
https://github.com/madflojo/hord/blob/main/hord.go#L100-L106

Reply:
I was debating whether to return the error at all considering the read has been successful and the cache write-back could be considered non-essential.

Reply:
Have wrapped in a ErrCacheError error.